code/go/0chain.net/chaincore/chain/state_handler.go

```diff
@@ -42,7 +85,12 @@ func (c *Chain) HandleSCRest(w http.ResponseWriter, r *http.Request) {
 		scRestRE = regexp.MustCompile(`/v1/screst/(.*)?/(.*)`)
 		pathParams = scRestRE.FindStringSubmatch(r.URL.Path)
 		if len(pathParams) == 3 {
-			common.ToJSONResponse(c.GetSCRestOutput)(w, r)
+			if !common.CheckCrossOrigin(w, r) {
```

Comment:
Would it be better if crossorigin checks are handled by explicit middlewares?

Reply:
I've moved this method from the `state_handler `to the `common `package, as it was before.